IBPSA/Fluid/HeatPumps/ModularReversible/BaseClasses/PartialReversibleRefrigerantMachine.mo

```diff
@@ -0,0 +1,633 @@
+  parameter Boolean use_intSafCtr=true
+    "=true to enable internal safety control"
+    annotation (Dialog(group="Safety control"), choices(checkBox=true));
+  replaceable parameter
+    IBPSA.Fluid.HeatPumps.ModularReversible.Controls.Safety.Data.Wuellhorst2021 safCtrPar
+    constrainedby
+    IBPSA.Fluid.HeatPumps.ModularReversible.Controls.Safety.Data.Generic
+    "Safety control parameters" annotation (Dialog(enable=use_intSafCtr,
+    group="Safety control"),
+      choicesAllMatching=true,
+      Placement(transformation(extent={{42,-18},{58,-2}})));
+  //Condenser
+  parameter Modelica.Units.SI.Time tauCon=30
+    "Condenser heat transfer time constant at nominal flow"
+    annotation (Dialog(tab="Condenser", group="Dynamics"));
+  parameter Modelica.Units.SI.Temperature TCon_nominal
+    "Nominal temperature of condenser medium"
+    annotation (Dialog(group="Nominal condition"));
+  parameter Modelica.Units.SI.TemperatureDifference dTCon_nominal
+    "Nominal temperature difference in condenser medium"
+    annotation (Dialog(group="Nominal condition", tab="Condenser"));
+  parameter Modelica.Units.SI.MassFlowRate mCon_flow_nominal
+    "Nominal mass flow rate of the condenser medium"
+    annotation (Dialog(
+      group="Nominal condition",
+      tab="Condenser"));
+
+  parameter Modelica.Units.SI.PressureDifference dpCon_nominal(displayUnit="Pa")
+    "Pressure drop at nominal mass flow rate"
+    annotation (Dialog(group="Flow resistance", tab="Condenser"));
+  parameter Real deltaMCon=0.1
+    "Fraction of nominal mass flow rate where transition to turbulent occurs"
+    annotation (Dialog(tab="Condenser", group="Flow resistance"));
+  parameter Boolean use_conCap=true
+    "=true if using capacitor model for condenser heat loss estimation"
+    annotation (Dialog(group="Heat Losses", tab="Condenser"),
+                                          choices(checkBox=true));
+  parameter Modelica.Units.SI.HeatCapacity CCon
+    "Heat capacity of the condenser"
+    annotation (Dialog(
+      group="Heat Losses",
+      tab="Condenser",
+      enable=use_conCap));
+  parameter Modelica.Units.SI.ThermalConductance GConOut
+    "Outer thermal conductance for condenser heat loss calculations"
+    annotation (Dialog(
+      group="Heat Losses",
+      tab="Condenser",
+      enable=use_conCap));
+  parameter Modelica.Units.SI.ThermalConductance GConIns
+    "Inner thermal conductance for condenser heat loss calculations"
+    annotation (Dialog(
+      group="Heat Losses",
+      tab="Condenser",
+      enable=use_conCap));
+
+  parameter Modelica.Units.SI.Density rhoCon=MediumCon.density(staCon_nominal)
+    "Condenser medium density"
+    annotation (Dialog(tab="Condenser", group="Medium properties"));
+  parameter Modelica.Units.SI.SpecificHeatCapacity cpCon=
+      MediumCon.specificHeatCapacityCp(staCon_nominal)
+    "Condenser medium specific heat capacity"
+    annotation (Dialog(tab="Condenser", group="Medium properties"));
```

Comment:
Why not making these `final`? What is the use case in which a user wants to change them from the default assignments?

Reply:
Using these functions raises the Non-literal value warning in Dymola. Overriding to e.g. 4184 removes the warning. See `IBPSA.Fluid.HeatPumps.ModularReversible.Examples.ModularReversible_OneRoomRadiator` for example

Reply:
See https://github.com/ibpsa/modelica-ibpsa/pull/1820 for a fix for this.